source/operand.cpp

```diff
@@ -110,33 +96,15 @@ spv_result_t spvOperandTableValueLookup(spv_target_env env,
     const auto beg = group.entries;
     const auto end = group.entries + group.count;
 
-    // We need to loop here because there can exist multiple symbols for the
-    // same operand value, and they can be introduced in different target
-    // environments, which means they can have different minimal version
-    // requirements. For example, SubgroupEqMaskKHR can exist in any SPIR-V
-    // version as long as the SPV_KHR_shader_ballot extension is there; but
-    // starting from SPIR-V 1.3, SubgroupEqMask, which has the same numeric
-    // value as SubgroupEqMaskKHR, is available in core SPIR-V without extension
-    // requirements.
     // Assumes the underlying table is already sorted ascendingly according to
     // opcode value.
-    const auto version = spvVersionForTargetEnv(env);
     for (auto it = std::lower_bound(beg, end, needle, comp);
          it != end && it->value == value; ++it) {
       // We consider the current operand as available as long as
-      // 1. The target environment satisfies the minimal requirement of the
-      //    operand; or
-      // 2. There is at least one extension enabling this operand; or
-      // 3. There is at least one capability enabling this operand.
-      //
-      // Note that the second rule assumes the extension enabling this operand
-      // is indeed requested in the SPIR-V code; checking that should be
-      // validator's work.
-      if ((version >= it->minVersion && version <= it->lastVersion) ||
-          it->numExtensions > 0u || it->numCapabilities > 0u) {
-        *pEntry = it;
-        return SPV_SUCCESS;
-      }
+      // it is in the grammar.  It might not be *valid* to use,
+      // but that should be checked by the validator, not by parsing.
+      *pEntry = it;
+      return SPV_SUCCESS;
```

Comment:
@dneto0 , this triggers a compile error when rolling into ANGLE:
https://chromium-review.googlesource.com/c/angle/angle/+/4769856
https://ci.chromium.org/ui/p/angle/builders/try/linux-test/15729/overview
../../third_party/vulkan-deps/spirv-tools/src/source/operand.cpp:102:43: error: loop will run at most once (loop increment never executed) [-Werror,-Wunreachable-code-loop-increment]
102 |          it != end && it->value == value; ++it) {
Could you please fix?
@jonahryandavis FYI

Reply:
Dawn is seeing the same issue

Reply:
I have posted a PR to fix this, https://github.com/KhronosGroup/SPIRV-Tools/pull/5373 and confirmed that it fixes the roll for Dawn